common/metrics/config.go

```diff
@@ -0,0 +1,22 @@
+package metrics
+
+// Config provides configuration for a Metrics instance.
+type Config struct {
+	// Namespace is the namespace for the metrics.
+	Namespace string
+
+	// HTTPPort is the port to serve metrics on.
+	HTTPPort int
+
+	// MetricsBlacklist is a list of metrics to blacklist. To determine the fully qualified metric name
+	// for this list, use the format "metricName:metricLabel" if the metric has a label, or just "metricLabel"
+	// if the metric does not have a label. Any fully qualified metric name that matches exactly with an entry
+	// in this list will be blacklisted (i.e. it will not be reported).
+	MetricsBlacklist []string
```

Comment:
Should metrics filtering be something that is taken care of by the application or the component that is collecting the metrics?

Reply:
When we were preparing for the (since abandoned) traffic generator, this seemed like a useful concept. I had added a large number of metrics, but there was concern that some of them were low bang for the buck (since it costs us $$$ to store them). We were planning on disabling a bunch of metrics with configuration changes, and turning them on in the future if we ever had an issue where they would be useful for debugging.

That being said, if people don't think this is a useful feature, it would be fairly straight forward to remove. What do others think?

Reply:
Yeah all I'm saying is that it's probably possible to filter the metrics, logs in a similar manner with the grafana agent.

This is pretty useful now though because we haven't figure out how to do that in the grafana agent. Regardless, if we want to save more money we would need to learn how to do it in the grafana agent because we're not always running applications that allow metrics filtering in this manner.

Reply:
Makes sense. I have removed the metrics blacklisting feature from this framework.